src/controllers/personsPublicRegisterNameController.ts

```diff
@@ -0,0 +1,78 @@
+import { formatValidationError, getPageProperties } from "../validations/validation";
+import { BASE_URL, CHECK_YOUR_ANSWERS, PERSONS_NAME_ON_PUBLIC_REGISTER, PERSONAL_CODE } from "../types/pageURL";
+import { Session } from "@companieshouse/node-session-handler";
+import { saveDataInSession } from "../utils/sessionHelper";
+import { ClientData } from "../model/ClientData";
+import { getPreviousPageUrl } from "../services/url";
+import { USER_DATA, MATOMO_BUTTON_CLICK, PREVIOUS_PAGE_URL } from "../utils/constants";
+import { selectLang, addLangToUrl, getLocalesService, getLocaleInfo } from "../utils/localise";
+
+export const get = (req: Request, res: Response, next: NextFunction) => {
+    const lang = selectLang(req.query.lang);
+    const locales = getLocalesService();
+    const session: Session = req.session as any as Session;
+    const clientData: ClientData = session.getExtraData(USER_DATA) ? session.getExtraData(USER_DATA)! : {};
+    const payload = {
+        "first-name": clientData.PublicRegisterName,
+        "middle-names": clientData.PublicRegisterMiddleName,
+        "last-name": clientData.PublicRegisterLastName
+    };
+
+    const previousPageUrl = getPreviousPageUrl(req, BASE_URL);
+    saveDataInSession(req, PREVIOUS_PAGE_URL, previousPageUrl);
+
+    const previousPage = previousPageUrl === addLangToUrl(BASE_URL + CHECK_YOUR_ANSWERS, lang)
+        ? addLangToUrl(BASE_URL + CHECK_YOUR_ANSWERS, lang)
+        : addLangToUrl(BASE_URL, lang);
+
+    res.render(config.PERSONS_NAME_ON_PUBLIC_REGISTER, {
+        previousPage: previousPage,
+        ...getLocaleInfo(locales, lang),
+        matomoButtonClick: MATOMO_BUTTON_CLICK,
+        currentUrl: BASE_URL + PERSONS_NAME_ON_PUBLIC_REGISTER,
+        payload
+    });
+};
+
+export const post = (req: Request, res: Response, next: NextFunction) => {
+    const locales = getLocalesService();
+    const lang = selectLang(req.query.lang);
+    const errorList = validationResult(req);
+    if (!errorList.isEmpty()) {
+        const pageProperties = getPageProperties(formatValidationError(errorList.array(), lang));
+        const session: Session = req.session as any as Session;
+
+        const previousPageUrl: string = session?.getExtraData(PREVIOUS_PAGE_URL)!;
+        const previousPage = previousPageUrl === addLangToUrl(BASE_URL + CHECK_YOUR_ANSWERS, lang)
+            ? addLangToUrl(BASE_URL + CHECK_YOUR_ANSWERS, lang)
+            : addLangToUrl(BASE_URL, lang);
+
+        res.status(400).render(config.PERSONS_NAME_ON_PUBLIC_REGISTER, {
+            ...getLocaleInfo(locales, lang),
+            previousPage: previousPage,
+            currentUrl: BASE_URL + PERSONS_NAME_ON_PUBLIC_REGISTER,
+            payload: req.body,
+            ...pageProperties
+        });
+    } else {
+        const session: Session = req.session as any as Session;
+        const clientData: ClientData = session.getExtraData(USER_DATA) ? session.getExtraData(USER_DATA)! : {};
```

Comment:
we should not need to handle the null client data like this as it should never be null on this page.
it should be const clientData: ClientData = session.getExtraData(USER_DATA)!